src/txn/txn_timestamp.c

```diff
@@ -207,15 +208,20 @@ __txn_get_pinned_timestamp(
 	conn = S2C(session);
 	txn_global = &conn->txn_global;
 
-	if (!txn_global->has_oldest_timestamp)
+	if (include_oldest && !txn_global->has_oldest_timestamp)
 		return (WT_NOTFOUND);
+
 	__wt_readlock(session, &txn_global->rwlock);
-	__wt_timestamp_set(tsp, &txn_global->oldest_timestamp);
+	if (include_oldest)
+		__wt_timestamp_set(tsp, &txn_global->oldest_timestamp);
+	else
+		__wt_timestamp_set_zero(tsp);
```

Comment:
I think better to use local variable for timestamp so that we will not modify the user passed parameter incase of returning WT_NOTFOUND.